playwright-tests/tests/profile.spec.js

```diff
@@ -0,0 +1,176 @@
+      );
+
+      await page.getByPlaceholder("Enter full name").fill("Someone");
+
+      const description = await page
+        .frameLocator("iframe")
+        .locator('textarea[name="textarea"]');
+      await description.fill("Someone");
+
+      await page.getByPlaceholder("eg. United States").fill("Someone");
+
+      await page.getByPlaceholder("twitter handle").fill("Someone");
+      await page.getByPlaceholder("github handle").fill("Someone");
+      await page.getByPlaceholder("telegram handle").fill("Someone");
+      await page.getByPlaceholder("website link").fill("Someone.com");
+
+      const expectedResult = {
+        "meghagoel.testnet": {
+          profile: {
+            name: "Someone",
+            description: "Someone",
+            image: {
+              ipfs_cid: "simple_cid",
+            },
+            location: "Someone",
+            linktree: {
+              twitter: "Someone",
+              github: "Someone",
+              telegram: "Someone",
+              website: "Someone.com",
+            },
+          },
+        },
+      };
+      // click button that triggers transaction
+      await page.getByRole("button", { name: "Save Profile" }).nth(0).click();
+      const transactionObj = JSON.parse(
+        await page.locator("div.modal-body code").innerText(),
+      );
+      // do something with transactionObj
+      expect(transactionObj).toMatchObject(expectedResult);
+    });
+  });
+
+  test.describe("User not logged in", () => {
+    test.use({
+      storageState: "playwright-tests/storage-states/wallet-not-connected.json",
+    });
+
+    test("Should show login prompt if no accountId is passed", async ({
+      page,
+    }) => {
+      await page.goto(`/${ROOT_SRC}?page=profile`);
+      const loginPrompt = page.getByText(
+        "AccountID prop or signed in account is required",
+      );
+      await expect(loginPrompt).toBeVisible();
+    });
+
+    test("Should show profile page when accountId is passed", async ({
+      page,
+    }) => {
+      await page.goto(`/${ROOT_SRC}?page=profile&accountId=efiz.testnet`);
+      const profileName = page.getByRole("heading", {
+        name: "Elliot",
+      });
+      await expect(profileName).toBeVisible();
+    });
+
+    test("Should navigate to Posts and Show them", async ({ page }) => {
+      await page.goto(`/${ROOT_SRC}?page=profile&accountId=efiz.testnet`);
+      const postButton = page.getByRole("tab", { name: "Posts" });
+      await postButton.click();
+
+      await expect(postButton).toHaveClass("nav-link active");
+    });
+    test("Should navigate to NFTs and Show them", async ({ page }) => {
+      await page.goto(`/${ROOT_SRC}?page=profile&accountId=efiz.testnet`);
+      const nftButton = page.getByRole("tab", { name: "NFTs" });
+      await nftButton.click();
+
+      await expect(nftButton).toHaveClass("nav-link active");
+    });
+    test("Should navigate to Widgets and Show them", async ({ page }) => {
```

Comment:
It looks like we're duplicating tests across "User logged in" and "User not logged in", although we don't expect any difference between the scenario's anyway.

I think "User logged in" -> "should show logged in user's profile page if no accountId is passed" is a good test, as is "User is not logged in" -> "Should show login prompt if no accountId is passed"

"User is logged in" -> "Should navigate to Edit Profile and save correct data" is a good test

I feel we are missing:
"User is not logged in" -> "should not be able to navigate to edit profile"
"User is logged in" -> "if not profile owner, should not be able to navigate to edit profile"

But tests to verify NFTs, Widget, posts don't need to be duplicated across the two.